### PR TITLE
Feature/nested mirage modules

### LINKED ISSUE #1414: Subdirectory / nested module support
6 upvotes from Canny

### PR DESCRIPTION
**Pull request by ro0gr**
Fixes #1414

### CODE REVIEW

**Comment by ro0gr** [comment]
@samselikoff please let me know if you have a better test setup suggestions in your mind

File: test-projects/01-basic-app/tests/acceptance/start-mirage-test.js [hunk 1]
```diff
@@ -52,6 +53,32 @@ module('Acceptance | Starting mirage', function(hooks) {
...
+
+        assert.ok(model instanceof NestedThingModel, 'models');
+        assert.equal(model.id, 'nested identity manager works!', 'identity managers');
+        assert.equal(model.name, 'nested factory works!', 'factories');
+
+        const { attributes } = server.serializerOrRegistry.serialize(model).data;
+        assert.ok('nested_thing_name' in attributes, 'serializer');
+      });
+
+      // factories and fixtures have to be tested separately
```
Comment (ro0gr) [inline test-projects/01-basic-app/tests/acceptance/start-mirage-test.js]: well, we could test it in a single test, but:
 - a nested identity manager does only support a single ID
 - it just looks cleaner to separate

**Comment by samselikoff** [comment]
I think this is good! There could be some more edge cases with ORM naming conventions, polymorphic associations etc. but I think this is a strict improvement over what we have!

**Comment by samselikoff** [comment]
Thanks so much for working on this. LMK if ready to merge, looks good on my end

**Comment by ro0gr** [comment]
I've been using this in a forked version for quite a while in few apps(no polymorphism though) and haven't noticed any issues. I think should be good to go.

**Comment by samselikoff** [comment]
Awesome – thanks again @ro0gr!
